02_cancer_type_classification/domain_adaptation.ipynb

```diff
@@ -0,0 +1,1567 @@
9Ba2+FepR77F44gPPiC0vFKrNjbWWfd/bo3xKmyK06VXbm6+SzxIdHEh8fiWaOSwgmzWFlZmbVgLeyZ2JCQYAAqV3migC24y9lgkW/Xb4YzbtpK9HoD5dzsSUxK5+iJQCZ8vYpvpw1WOl6O9Dodayd+wr3rV0zbTq7/i7uBF3lx0lxU6nxNmfTYci1aDx8+/NC2kydPMnHiRBYuXGiWMEIIIYTIXWkYm1VqDRqtFXpdJqkJkTh5eJEafw8ArXXhzwaWVY4eLkAYt/6+QYUGVYgLiSbhdiyowLFc2blXuLBng6dNmwjA2LFfFkEaIR5t47ZT6PUGenapx7gPnyEqJon+wxYRcCGEG8H38PWpoHTEhwSd/Jt716/g4FGBjmOmY9Tr2TPtU8IunSPk7El8mjxpln5zLVpz0rx5czIzM/N8nsFgYMKECVy9ehVra+uHFkDfvHkzf/zxx7/3NPgxYcIE1GaqyoUQQojSrKSNzWq1Bp+6bbhxbj9Hlo/BrXJtom9lnWGtVq9tkfdXVtTp3Iigo1cIORTIrX9uYMjMWkqieqta2Ct0j9y9e+HcuRNGzZp+ODqWncJZiPyKjkkE4Mmm3qjVKjw9nKju7cHZi2FERSdaZNEaefM6AL7tn6FCnYYAVPPvzKXNK4i8ed1sRetjj0ZJSUl5Pmf37t1kZGSwYsUKPvroI6ZPn27al5aWxty5c1m8eDHLly8nKSmJffv2PW4MIYQQQvyrpI3NTdq/QkWfBugyUogMPo1Bn4F3nVbUfaqX2fos7dy9ytPpg164VSmHIVOP1lpL7U4NafV6p2LPkpyczPDhb9KwYS26detA/fp+zJ07C+O/swYLIbLUqZV1GfofK49zMiCUVZvOcO5SGFqtGj9fy7ik/79cPLNy3T71N2kJcaTGxRAW8A8AzhXMlznflwdnZGSwc+dOmjVrlmejp06dom3brE9LGzduzIUL/3+PirW1NcuXL8fOLusGd51Oh42NTYHCCyGEEGVJaRmbrWzs6fDip8RGhJAUew+X8lVxdrfMN2glSeV6XvSaNJCMlHQ01lo0Wo0iOb744lPWrl2NlVrNEw6OBCclMnXqJLp1e5a6desqkkkIS9T72eZs3XWG6zejGP7pStP2l/u2sdiJmGq2bsfRFQuJuXmNpa92BSMYDXpcKlbG98mimak8J7kWrVu2bMn22NbWFj8/P1JTU/NsNCkpCUfH/78URaPRoNPp0Gq1qNVqPDw8AFiyZAkpKSm0afPob1CjUeHqar7ZqIQQQoiSwFLHZq22YJcRu3l64+ZZdJPWaLXqXN8v/Pbbr9y8GVTgtkNDgwGYOXNygduoVq06Q4YMzXFfQV/DnFjbF/wDh0e9hnnZu3cv27Zt4a+//kQF/Nq2AzVcXFhz8wZzzp/lxIljODraP/I17NSpCx07dixg+sK7/3Ow1Pedlp4PJOOD7efFydGOeTPe4q81Rzh97iZOjnZ079KYzu0aFEmG3L6/wvy9sbaz54WJ37D3128JCTgBKhXVmj5Fh7f/h9bKusgyPvTc3HZMmzbN9PW5c+f4888/WbBgAd26dcuzUUdHR5KTk02PDQYDWq022+OZM2dy8+ZNvv/+e1SqRy96rdcbiYtLybNfIYQQIj/Kly+Z99dZ6tis0xke91sxC53OkOv7hWvXrhN87TIVHQs2c629Mese0aRbgQU6Pjwp85H5SsJrmJeUlHRSUlIxGAw4WllRzdkZgAbu5QDIzMzExcX1kd9rSkq6ou/57mez1Pedlp4PJOOD7eeHm6sDI97qapYM5vp741qpCn3HzyQjNav9gi5189+Mjxqbcy1aMzIy2LJlC8uWLcPKyoqkpCT27NmDrW3eM/s1bdqUffv20b17dwICAvDz88u2f/z48VhbW/Pjjz/KBExCCCFEPsnYXDgVHa14q4mHIn0vOBOlSL/Fyd+/Ha1bt2XHjp2EhgYz53wAbStW5s9rVwHo3r0n06fPUTilEKKomHNd1v/KtWjt2LEjzz33HDNnzsTHx4chQ4bka1AE6NKlC0eOHGHAgAEYjUamTp3Kpk2bSElJoX79+qxevZrmzZvz+uuvA/Daa6/RpUuXovmOhBBCiFJKxmZh6dRqNRMnfsVbbw1iffBN1gffBMDdzZ1Roz5ROJ0QoqTKtWh97bXX2Lx5M2FhYbzwwguPNeObWq1m0qRJ2bb5+vqavr5y5cp/DxFCCCFEHmRsFiVBjx492bRpBwsW/Mzt27dp1Kgxw4ePpGpVL6WjCSFKqFyL1rfffpu3336b48ePs2rVKi5cuMDMmTPp3bv3Q5cUCSGEEML8LHls1qfFkhi0q0DHGnRZE0mptXaF6h+UufRXPKxFi5a0aNFS6RhCiFIi16L1vieffJInn3yShIQENmzYwOjRo1m/fn0xRBNCCCFETixtbPbyKtwMwCEhwQB4exWm6PQodA4hhBCWKc+i9T5nZ2cGDRrEoEGDzJlHCCGEEPlkKWPzwIGDC3X8tGkTARg79ssiSFO0ohNT+X1vAP9cC8Pe2ooezWrwYqs6aErpZFVCiLIhPTqCkM3LCny8LiVrNnqtfcHWk02PjgCX/M/kn++iVQghhBCiLElKy2DI/M3cik4wbbscFsWN8Fg+7+evYDIhhCi4orgqJSQuGgDvShUL1oCL02PlkKJVCCGEECIHG08Ecis6gZqezkzt04SwuBQ+WXOKdcev8nr7hlQp56x0RCGEeGyFvToGiv8KGbm2RQghhBAiB5fDstZWfe2p6jT1KkfPhlVp7VsegCth0UpGE0KIMkWKViGEEEKIHFRwybpX60DgPTJ0Bu7Gp3D+diwAFd0clYwmhBBlilweLIQQQgiRg+db1mb5kUvsvnKXFtM2k5apR2cw0tDLk3pVZHkdIYQoLlK0CiGEEELk4Al3J757sytfbzjKjfBYVCroUM+bsX3boFKplI4nhLBgEVEpLF1/uUDHJqdkAuBgb1XgvquVslvupWgVQgghhMhF0
```

Comment:
I'm surprised that the values of AUPR(signal) - AUPR(shuffled) would be consistently higher than 0.5, because my intuition tells me that AUPR(shuffled) should be around 0.5. Am I thinking about this wrong?

Reply:
AUROC is usually around 0.5 for a (binary) classifier that guesses randomly, but AUPR can be considerably lower depending on the proportion of positive labels (see, e.g. [this post](https://stats.stackexchange.com/a/266989) for more detail).

Looks like it's around 0.1-0.2 for CDKN2A, for most of the baseline models.